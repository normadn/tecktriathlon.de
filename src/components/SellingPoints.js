import React from "react";
import SellingPointsIcons from "./SellingPointsIcons.js";

import cubes2png from "../assets/img/icons/cubes-2.png";
import judges from "../assets/img/icons/judgesIcon.svg";
import cubes2webp from "../assets/img/icons/cubes-2.webp";
import mentors from "../assets/img/icons/mentorsIcon.svg";
import hackers from "../assets/img/icons/hackersIcon.svg";
import schools from "../assets/img/icons/schoolsIcon.svg";
import challenges from "../assets/img/icons/challengesIcon.svg";
import applications from "../assets/img/icons/applicationsIcon.svg";
import qhacksSellingPointOnejpg from "../assets/img/selling_points/sellingPointElement.jpg";
import qhacksSellingPointTwojpg from "../assets/img/selling_points/sellingPointCode.jpg";
import qhacksSellingPointOnewebp from "../assets/img/selling_points/qhacksSellingPointOne.webp";
import qhacksSellingPointTwowebp from "../assets/img/selling_points/qhacksSellingPointTwo.webp";
import ContentWrapper from "./ContentWrapper";

const twoColumnStyle = {
  display: "flex",
  justifyContent: "center",
  alignItems: "center",
  margin: "0 auto",
  marginBottom: "48px"
};

const flexChildStyle = {
  flex: 1,
  textAlign: "center",
  minWidth: "300px",
  margin: "0 28px 32px 28px",
  maxWidth: "510px"
};

const blurbStyle = {
  margin: "32px auto",
  fontSize: "15px",
  lineHeight: "24px",
  maxWidth: "500px"
};

const imgStyle = {
  maxHeight: "320px",
  maxWidth: "475px",
  width: "100%"
};

const SellingPoints = () => (
  <div id="services" css={{ position: "relative" }}>
    <ContentWrapper>
      <div
        css={{
          padding: "143px 0",
          margin: "0 -28px",
          "@media(max-width: 840px)": {
            padding: "143px 0 50px 0"
          }
        }}
      >
        <div
          css={{
            ...twoColumnStyle,
            flexWrap: "wrap"
          }}
        >
          <div css={flexChildStyle}>
            <h1>Softwareelemente</h1>
            <p css={blurbStyle}>
              Sie wollen ihre Unternehmens-Website erneuern, ihr SEO-Ranking verbessern, ihre Kundendaten aufarbeiten oder Prozesse automatisieren? Wir haben Spezialisten für im Frontend, Backend, Web-Design und der Datenanalyse im Team und können Softwareelemente auf Ihr Unternehmen zugeschnitten entwickeln.  
            </p>
          </div>
          <div css={flexChildStyle}>
            <picture css={imgStyle}>
              <source srcSet={qhacksSellingPointOnejpg} type="image/jpg" />
              <img
                css={imgStyle}
                src={qhacksSellingPointOnejpg}
                alt="html css"
              />
            </picture>
          </div>
        </div>
        <div
          css={{
            ...twoColumnStyle,
            flexWrap: "wrap-reverse",
            marginBottom: 0
          }}
        >
          <div css={flexChildStyle}>
            <picture css={imgStyle}>
              <source srcSet={qhacksSellingPointTwojpg} type="image/jpg" />
              <img
                css={imgStyle}
                src={qhacksSellingPointTwojpg}
                alt="Mentor judging a VR hack"
              />
            </picture>
          </div>
          <div css={flexChildStyle}>
            <h1>Komplettlösungen</h1>
            <p css={blurbStyle}>
              Sie suchen nach einer Digitalisierungsstrategie für ihr Unternehmen? - Da sind sie bei uns genau richtig! Jedes Unternehmen löst andere Herausforderungen. Ein enger Austausch mit Ihnen ist uns folglich wichtig, um Lösungen zu entwickeln, die Ihre Kunden begeistern werden. Wir helfen Ihnen der Konkurrenz einen Schritt voraus zu sein.
            </p>
          </div>
        </div>
        <picture
          css={{
            width: "400px",
            height: "246px",
            position: "absolute",
            bottom: "-140px",
            left: "-60px",
            zIndex: 3,
            "@media(max-width: 1200px)": {
              width: "358px",
              height: "220px",
              bottom: "-120px",
              left: "-40px"
            },
            "@media(max-width: 840px)": {
              display: "none"
            }
          }}
        >
          <source srcSet={cubes2webp} type="image/webp" />
          <source srcSet={cubes2png} type="image/png" />
          <img
            css={{
              width: "400px",
              height: "246px",
              "@media(max-width: 1200px)": {
                width: "358px",
                height: "220px"
              }
            }}
            src={cubes2png}
            alt="Floating cubes"
          />
        </picture>
      </div>
    </ContentWrapper>
  </div>
);

export default SellingPoints;
