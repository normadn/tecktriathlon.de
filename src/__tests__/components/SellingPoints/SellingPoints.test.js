import React from "react";
import SellingPoints from "../../../components/SellingPoints";
import { shallow } from "enzyme";

describe("SellingPoints", () => {
  it("renders correctly", () => {
    expect(shallow(<SellingPoints />)).toMatchSnapshot();
  });
});
