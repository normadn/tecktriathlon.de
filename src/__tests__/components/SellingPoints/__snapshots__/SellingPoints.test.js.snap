// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SellingPoints renders correctly 1`] = `
<div
  css={
    Object {
      "position": "relative",
    }
  }
>
  <_default>
    <div
      css={
        Object {
          "@media(max-width: 840px)": Object {
            "padding": "143px 0 50px 0",
          },
          "margin": "0 -28px",
          "padding": "143px 0",
        }
      }
    >
      <div
        css={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexWrap": "wrap",
            "justifyContent": "center",
            "margin": "0 auto",
            "marginBottom": "48px",
          }
        }
      >
        <div
          css={
            Object {
              "flex": 1,
              "margin": "0 28px 32px 28px",
              "maxWidth": "510px",
              "minWidth": "300px",
              "textAlign": "center",
            }
          }
        >
          <h1>
            Grow Your Network.
          </h1>
          <p
            css={
              Object {
                "fontSize": "15px",
                "lineHeight": "24px",
                "margin": "32px auto",
                "maxWidth": "500px",
              }
            }
          >
            With hundreds of talented, ambitious, and like-minded individuals all in one place, QHacks is the perfect playground for any tech enthusiast! Over the weekend, immerse yourself in the community, meet new people and talk shop with other students and companies! Strong connections aren’t only important for internet speeds - this is your chance to build, learn and grow!
          </p>
          <SellingPointsIcons
            list={
              Array [
                Object {
                  "icon": "test-file-stub",
                  "number": 1800,
                  "text": "Applications",
                },
                Object {
                  "icon": "test-file-stub",
                  "number": 25,
                  "text": "Schools",
                },
                Object {
                  "icon": "test-file-stub",
                  "number": 500,
                  "text": "Hackers",
                },
              ]
            }
          />
        </div>
        <div
          css={
            Object {
              "flex": 1,
              "margin": "0 28px 32px 28px",
              "maxWidth": "510px",
              "minWidth": "300px",
              "textAlign": "center",
            }
          }
        >
          <picture
            css={
              Object {
                "maxHeight": "320px",
                "maxWidth": "475px",
                "width": "100%",
              }
            }
          >
            <source
              srcSet="test-file-stub"
              type="image/webp"
            />
            <source
              srcSet="test-file-stub"
              type="image/jpg"
            />
            <img
              alt="Hackers in a workshop"
              css={
                Object {
                  "maxHeight": "320px",
                  "maxWidth": "475px",
                  "width": "100%",
                }
              }
              src="test-file-stub"
            />
          </picture>
        </div>
      </div>
      <div
        css={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexWrap": "wrap-reverse",
            "justifyContent": "center",
            "margin": "0 auto",
            "marginBottom": 0,
          }
        }
      >
        <div
          css={
            Object {
              "flex": 1,
              "margin": "0 28px 32px 28px",
              "maxWidth": "510px",
              "minWidth": "300px",
              "textAlign": "center",
            }
          }
        >
          <picture
            css={
              Object {
                "maxHeight": "320px",
                "maxWidth": "475px",
                "width": "100%",
              }
            }
          >
            <source
              srcSet="test-file-stub"
              type="image/webp"
            />
            <source
              srcSet="test-file-stub"
              type="image/jpg"
            />
            <img
              alt="Mentor judging a VR hack"
              css={
                Object {
                  "maxHeight": "320px",
                  "maxWidth": "475px",
                  "width": "100%",
                }
              }
              src="test-file-stub"
            />
          </picture>
        </div>
        <div
          css={
            Object {
              "flex": 1,
              "margin": "0 28px 32px 28px",
              "maxWidth": "510px",
              "minWidth": "300px",
              "textAlign": "center",
            }
          }
        >
          <h1>
            Showcase to Experts.
          </h1>
          <p
            css={
              Object {
                "fontSize": "15px",
                "lineHeight": "24px",
                "margin": "32px auto",
                "maxWidth": "500px",
              }
            }
          >
            This year, we’re bringing together a diverse group of mentors, speakers, sponsors and other industry professionals. Chat one-on-one, learn about current tech trends and opportunities, and show off your skills! Looking for some inspiration? Encountering a roadblock? Does python feel like it’s constricting you? Not to worry, there will always be a helping hand at the ready.
          </p>
          <SellingPointsIcons
            list={
              Array [
                Object {
                  "icon": "test-file-stub",
                  "number": 40,
                  "text": "Mentors",
                },
                Object {
                  "icon": "test-file-stub",
                  "number": 20,
                  "text": "Judges",
                },
                Object {
                  "icon": "test-file-stub",
                  "number": 15,
                  "text": "Challenges",
                },
              ]
            }
          />
        </div>
      </div>
      <picture
        css={
          Object {
            "@media(max-width: 1200px)": Object {
              "bottom": "-120px",
              "height": "220px",
              "left": "-40px",
              "width": "358px",
            },
            "@media(max-width: 840px)": Object {
              "display": "none",
            },
            "bottom": "-140px",
            "height": "246px",
            "left": "-60px",
            "position": "absolute",
            "width": "400px",
            "zIndex": 3,
          }
        }
      >
        <source
          srcSet="test-file-stub"
          type="image/webp"
        />
        <source
          srcSet="test-file-stub"
          type="image/png"
        />
        <img
          alt="Floating cubes"
          css={
            Object {
              "@media(max-width: 1200px)": Object {
                "height": "220px",
                "width": "358px",
              },
              "height": "246px",
              "width": "400px",
            }
          }
          src="test-file-stub"
        />
      </picture>
    </div>
  </_default>
</div>
`;
